autogluon/utils/tabular/ml/models/fastainn/tabular_nn_fastai.py

```diff
@@ -0,0 +1,334 @@
+            'f1_weighted': FBeta(beta=1, average='weigthed'),  # this one has some issues
+
+            'roc_auc': AUROC(),
+
+            'precision': Precision(),
+            'precision_macro': Precision(average='macro'),
+            'precision_micro': Precision(average='micro'),
+            'precision_weigthed': Precision(average='weigthed'),
+
+            'recall': Recall(),
+            'recall_macro': Recall(average='macro'),
+            'recall_micro': Recall(average='micro'),
+            'recall_weigthed': Recall(average='weigthed'),
+            # Not supported: pac_score
+        }
+
+        objective_func_name = self.eval_metric.name
+        if objective_func_name in metrics_map.keys():
+            nn_metric = metrics_map[objective_func_name]
+        elif objective_func_name is None:
+            objective_func_name = self.params['metric']
+            nn_metric = metrics_map[self.params['metric']]
+        else:
+            nn_metric = None
+        return nn_metric, objective_func_name
+
+    def __get_objective_func_to_monitor(self, objective_func_name):
+        monitor_obj_func = {
+            'roc_auc': 'auroc',
+
+            'f1': 'f_beta',
+            'f1_macro': 'f_beta',
+            'f1_micro': 'f_beta',
+            'f1_weighted': 'f_beta',
+
+            'precision_macro': 'precision',
+            'precision_micro': 'precision',
+            'precision_weigthed': 'precision',
+
+            'recall_macro': 'recall',
+            'recall_micro': 'recall',
+            'recall_weigthed': 'recall',
+            'log_loss': 'valid_loss',
+        }
+        objective_func_name_to_monitor = objective_func_name
+        if objective_func_name in monitor_obj_func:
+            objective_func_name_to_monitor = monitor_obj_func[objective_func_name]
+        return objective_func_name_to_monitor
+
+    def predict_proba(self, X, preprocess=True):
+        from fastai.basic_data import DatasetType
+        from fastai.tabular import TabularList
+        from fastai.utils.mod_display import progress_disabled_ctx
+        from fastai.tabular import FillMissing, Categorify, Normalize
+
+        if preprocess:
+            X = self.preprocess(X)
+        procs = [FillMissing, Categorify, Normalize]
+        self.model.data.add_test(TabularList.from_df(X, cat_names=self.cat_columns, cont_names=self.cont_columns, procs=procs))
+        with progress_disabled_ctx(self.model) as model:
+            preds, _ = model.get_preds(ds_type=DatasetType.Test)
+        if self.problem_type == REGRESSION:
+            if self.y_scaler is not None:
+                return self.y_scaler.inverse_transform(preds.numpy()).reshape(-1)
+            else:
+                return preds.numpy().reshape(-1)
+        if self.problem_type == BINARY:
+            return preds[:, 1].numpy()
+        else:
+            return preds.numpy()
+
+    def save(self, file_prefix='', directory=None, return_filename=False, verbose=True):
```

Comment:
Just to make sure: You aren't pickling the neural network weights right? These will be saved in torch format?

Reply:
It's more than weights that are saved here: weights + data loaders state + classes mappings + optimizer states. fast.ai bundles everything needed for prediction into this object, then we pickle this export object.